.github/actions/buildcocoasdk/action.yml

```diff
@@ -0,0 +1,45 @@
+name: Build Native Dependencies
+runs:
+  using: composite
+
+  steps:
+
+    - name: Get Carthage SHA
+      shell: bash
+      run: echo "CARTHAGE_SHA=$(git rev-parse HEAD)" >> $GITHUB_ENV
+      working-directory: modules/Carthage
+
+    - name: Cache Carthage
+      id: cache-carthage
+      uses: actions/cache@v3
+      with:
+        path: modules/Carthage/.build/debug/carthage
```

Comment:
can we go without this cache since we're caching below at `Cache Sentry Cocoa SDK`?

Reply:
I split them into separate caches because Carthage takes a very long time to build, and it's not likely we'll need to update it as often as we do the Cocoa SDK.  I also restore it every time to keep it active, otherwise GHA will prune it from the cache over time.